int32 id1
int32 id2
float32[6] transformation
